pom.xml

```diff
@@ -37,6 +36,32 @@
     <module>distribution</module>
   </modules>
 
+  <profiles>
+    <profile>
+      <id>jdk8</id>
+      <activation>
+        <jdk>[,9)</jdk>
+      </activation>
+      <properties>
+        <xalan.taglet.artifactId>xalan2jtaglet</xalan.taglet.artifactId>
+      </properties>
+      <modules>
+        <module>xalan2jtaglet</module>
+      </modules>
+    </profile>
+    <profile>
+      <id>jdk9+</id>
+      <activation>
+        <jdk>[9,)</jdk>
+      </activation>
+      <properties>
+        <xalan.taglet.artifactId>xalan2jtaglet_jdk9</xalan.taglet.artifactId>
+      </properties>
+      <modules>
+        <module>xalan2jtaglet_jdk9</module>
+      </modules>
+    </profile>
+  </profiles>
```

Comment:
Even though adding two profiles is not much code, I suggest just requiring Java 17 for build, and use `--release 8`, `--target 8` to target Java 8 bytecode. It would not require profiles, it would be easier to understand.

It would be way easier to support, and it would be way fewer bugs since javac 17 should be tested way better than javac 8.

Reply:
I had discussed that with @kubycsolutions in .

I said:
>> Another question is if you really want to force every developer to run the build on JDK 8 just because of the taglet dependencies on old `com.sun` JDK packages, which have been superseded by others on JDK 9+. Basically, you have 3 options:
>> 
>> * Continue as is.
>> * Require JDK 9+ for the build and migrate the taglet thingy to the new API standard.
>> * Add a JDK 9+ variant of the taglet code on top of the JDK 8 version and configure the build to use one on JDK 8 and the other on JDK 9+.

To which Joseph replied:

> Xalan currently promises to run on JRE 8. That doesn't absolutely require that we build on JDK 8, I suppose, but I want to retain the ability to do so. For now, that rules out requiring 9+. But we'll move forward someday.

As a first-time contributor to this project, I complied, which is the reason why the solution looks like this and no different. Vladimir @vlsi, can you please discuss with Joseph? My suggestion is to keep the build on JDK 8, because all the work is done already, and I would be quite disappointed to have invested my time just to revert most of it again now.

Reply:
My take is that dropping JDK8 build compatibility is a separate discussion, worth considering but a bit of a distraction at this moment. 

Changing the Java 8 compilation requirement -- if I'm not hallucinating that the policy still exists; remember I'm just getting back to Xalan 15 years later and still catching up -- is something I'd want to warn our users of. If nothing else, supporting both gives us a deprecation window.

So my take is: Valid question, open it separately as technical debt, discuss, and deal with it as its own issue rather than here.

Other Committers may have other opinions. But I'm going to accept this into my branch; it can always be simplified after that discussion, or folks can throw rocks at it before merge into main if they really feel we need to delay the already-large mvn proposal further. 

(FWIW, I'm inclined to withdraw my PR to Master/Main again. It was originally posted for discussion; it wasn't ever ready for merge and in hindsight I should have just asked folks to review it unsquashed on the fork. And in fact once we have it ready to consider  merging, I'm tempted to reconstruct it in proper incremental change form, thus undoing the squash and making the actual concepts clearer in the history, at the admitted cost of losing some history of my mistakes. Downside is my being distracted by it that much longer. I freely admit I mis-sized the effort to learn the Maven Way.)

Reply:
> My take is that dropping JDK8 build compatibility is a separate discussion

Agreed.

> Changing the Java 8 compilation requirement -- if I'm not hallucinating that the policy still exists

If it still exists, it is a bit weird. Java 8 is no longer supported, even as an LTS release. I understand that Xalan should _run_ on Java 8, but where is the value in _building_ it on Java 8? Anyway, I implemented it that way and hope you can merge it soon. Sorry for the pressure, but I always try to minimise the difference between touch time and cycle time in everything I do. Frequent context changes lead to loss of efficiency. Moreover, code rots.

> If nothing else, supporting both gives us a deprecation window.

True.

> (FWIW, I'm inclined to withdraw my PR to Master/Main again. It was originally posted for discussion; it wasn't ever ready for merge

Then mark it as a draft PR.

![image](https://github.com/apache/xalan-java/assets/1537384/4c79e109-e3b6-48b7-b818-0ef6cc870398)

> once we have it ready to consider merging, I'm tempted to reconstruct it in proper incremental change form

That would be a total waste of time. It's what it is (sorry, I watched "The Irishman", could not resist). That PR was open too long already. I suggest focusing on getting it done, merged and leave Ant behind ASAP.

Moreover, I would not like to see my commits being redone by someone else who is then listed as an author.